app/controllers/songs_controller.rb

```diff
@@ -16,10 +16,10 @@ def index
         songs = songs.search_by_keywords(search_value) if search_value.present?
         songs = songs.where(key: params[:key]) if params[:key].present?
         songs = songs.where(tempo: params[:tempo]) if params[:tempo].present?
+        # get number of matching songs post filtering
+        records_filtered = songs.count
```

Comment:
Solution to the problem below: use `count`, which issues a simple count SQL query. Note I tried to put it on line 22 but the select statement messes up the SQL generated the Postgres errors out.